# How to find Cosine Similarity between two numeric array in Python

### Cosine similarity measures the similarity between two vectors. It is measured by the cosine of the ange between those two vectors.

In [1]:
#Here we will see two ways of finding the Cosine Similarity between two array.


### METHOD 1:

In [2]:
#Import Library

import numpy as np

from numpy.linalg import norm
from numpy import dot

In [3]:
# Create an example arrays


a = [2, 3, 4, 5, 6]
b = [7, 8, 2, 9, 3]


In [4]:
cos_sim = dot(a, b)/(norm(a)*norm(b))

In [5]:
cos_sim

0.798583017779342

From above value we can say that there is 79% similarity between a and b 

### METHOD 2:

In [6]:
# import cosine similarity measure 


from sklearn.metrics.pairwise import cosine_similarity

In [7]:
# Create an example arrays

arr1=np.array([[1,1,2,3,1]])
arr2=np.array([[1,1,2,4.5,1]])

In [8]:
# Check the cosine similarity between aar1 and arr2 

cosine_similarity_array= cosine_similarity(arr1,arr2)
cosine_similarity_array

array([[0.98177194]])

From above value we can say that there is 98% similarity between arr1 and arr2 

### ----------------------------------------------------Question 12 Ends -------------------------------------------------------